api/include/opentelemetry/nostd/absl/README.md

```diff
@@ -0,0 +1,9 @@
+# Notes on Abseil Variant implementation
+
+This is a snapshot of Abseil Variant `absl::variant` from Abseil `v2020-03-03#8`.
+
+This code is required to compile OpenTelemetry code in vs2015 because MPark Variant implementation is not compatible with vs2015.
+
+Build option `HAVE_ABSEIL_VARIANT` allows to enable the build with `absl::variant`, `absl::get` and `absl::visit` as defalt implementation for `nostd::` classes.
```

Comment:
Is there a reason we have to ship two variant implementations? If `absl` is needed for VS2015 (and we decide we want to support that), can't we just generally use the `absl` implementation (instead of the MPark one)?

Reply:
I'd like to take a two-staged approach. We should use Abseil everywhere, but I'd like to keep it under the build option right now until I fully test all existing exporters with it. Within the same OS class the ABI compatibility is retained irrespective of what implementation we take, e.g. Mpark works for Linux and Mac no problem. Whereas Abseil works for Windows. I want to send a follow-up PR that adds a build loop for Windows to switch over to Abseil for Variant. Then we can do this for Linux and Mac.

Reply:
I think long-term before v1.0 GA we should switch to Abseil for variant. That'd be great for another reason: we can get rid of Boost license in our list of licenses because only MPark variant is Boost License, whereas Abseil is also Apache License v2.0 .. but I'd like to do this switch in a separate PR.

Reply:
#68

Reply:
#68

Reply:
Makes sense. Can you add an additional issue for this (apart from the license one), otherwise this might be lost when just looking at issue titles.

I'd be fine with switching to absl as default, having MPark under a switch (then our CI pipeline would verify how it's working). However, I assume you expect changes to be necessary for it to work on Linux.

Reply:
Have created an issue for this #572 , and assigned to @maxgolov . Feel free to update with relevant details on it.